File: s-recursos/src/main/java/co/edu/javeriana/arqui/rest/controller/ClinicaController.java
package co.edu.javeriana.arqui.rest.controller;

import java.util.List;

import co.edu.javeriana.arqui.rest.model.Clinica;
import co.edu.javeriana.arqui.rest.service.ClinicaService;
import jakarta.inject.Inject;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/clinicas")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class ClinicaController {

    @Inject
    ClinicaService service;

    @POST
    public Clinica registrar(Clinica c) {
        return service.registrar(c);
    }

    @GET
    public List<Clinica> listar() {
        return service.listar();
    }
}

File: ms-clinicas/src/main/kotlin/co/edu/javeriana/arqui/controller/ClinicaController.kt
package co.edu.javeriana.arqui.controller

import co.edu.javeriana.arqui.model.Clinica
import co.edu.javeriana.arqui.repository.ClinicaRepository
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response

@Path("/clinicas")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
@ApplicationScoped
class ClinicaController(private val repository: ClinicaRepository) {

    @GET
    fun getAllClinicas(): List<Clinica> = repository.findAll()

    @POST
    fun createClinica(clinica: Clinica): Response {
        println("Recibiendo clinica" + clinica.nombre + ", " + clinica.direccion)
        repository.save(clinica)
        return Response.status(Response.Status.CREATED).entity(clinica).build()
    }
}
